lib/msf/core/exploit/auto_check.rb

```diff
@@ -16,10 +16,24 @@ def initialize(info = {})
     ])
   end
 
+  def run
+    with_prepended_auto_check do
+      super
+    end
+  end
+
   def exploit
+    with_prepended_auto_check do
+      super
+    end
+  end
```

Comment:
I think this is a good first step for now; but long term I'd be in favor of replacing exploit module's `exploit` method with `run` and creating a separate ticket for completing effort in the new year :+1: